### PR TITLE
add emoji and mathjax support

### PR DESCRIPTION
**Pull request by thuleqaid**
Feature Added:
+ emoji support
+ mathjax support(local katex lib)

Modification:
+ make sure loading `open-browser.vim` before call `openbrowser#load()`
+ prepend `python` before command `rst2html.py` in win32 system

### CODE REVIEW

File: autoload/previm.vim [hunk 1]
```diff
@@ -208,7 +213,11 @@ function! s:do_external_parse(lines) abort
   endif
   let temp = tempname()
   call writefile(a:lines, temp)
-  return split(s:system(cmd . ' ' . s:escape_backslash(temp)), "\n")
+  if has('win32')
+    return split(s:system('python ' . exepath(cmd) . ' ' . s:escape_backslash(temp)), "\n")
+  else
+    return split(s:system(cmd . ' ' . s:escape_backslash(temp)), "\n")
+  endif
```
Comment (tyru) [inline autoload/previm.vim:220]: Instead of changing these lines, how about changing
```vim
let candidates = ['rst2html.py', 'rst2html']
```
to
```vim
let candidates = ['python rst2html.py', 'python rst2html']
```

But I thought previm works also on windows.
@mattn is this change necessary?

File: autoload/previm.vim [hunk 2]
```diff
@@ -40,6 +40,11 @@ endfunction
 
 function! s:exists_openbrowser() abort
   try
+    if get(g:, 'spacevim_plugin_manager', '') ==# 'dein'
+      if get(dein#get('open-browser.vim'), 'sourced', 0) == 0
+        call dein#source('open-browser.vim')
+      endif
+    endif
```
Comment (tyru) [inline autoload/previm.vim:47]: * Why did you add these lines? open-browser.vim is not loaded properly?
* These codes depend on SpaceVim and dein, is it really necessary?

File: autoload/previm.vim (hunk 1 above)
Comment (mattn) [reply autoload/previm.vim:220]: >is this change necessary?

No needed. It should work as tyru pointed.

**Comment by tyru** [comment]
emoji and mathjax support seems good, but I don't think `*.vim` changes are necessary.

**Comment by Bakudankun** [comment]
There are three PRs to add TeX support.
#41 has been suspended for its size.
#105 has gotten no reaction yet.
Is this PR to be adopted? (I don't have objections)

FYI this PR will increase the size of cloned repo from 6.6MB to 9.4MB, and also the working copy from 2.5MB to 13.6MB.

**Comment by mattn** [comment]
@thuleqaid can you provide URLs that you get the files? I want to provide a way to update dependencies.

**Comment by thuleqaid** [comment]
> @thuleqaid can you provide URLs that you get the files? I want to provide a way to update dependencies.

https://github.com/KaTeX/KaTeX
https://github.com/markdown-it/markdown-it-emoji
https://github.com/classeur/markdown-it-mathjax

https://github.com/markdown-it/markdown-it-ins
https://github.com/markdown-it/markdown-it-mark
https://github.com/RedBug312/markdown-it-multimd-table

https://github.com/markdown-it/markdown-it-container

**Comment by k-takata** [comment]
> * prepend `python` before command `rst2html.py` in win32 system

May I ask the reason?
The official python installer associates `*.py` to `py.exe` instead of `python.exe` and doesn't add `python.exe` to PATH by default. So, prepending `python` breaks the default installation.

**Comment by thuleqaid** [comment]
> > * prepend `python` before command `rst2html.py` in win32 system
> 
> May I ask the reason?
> The official python installer associates `*.py` to `py.exe` instead of `python.exe` and doesn't add `python.exe` to PATH by default. So, prepending `python` breaks the default installation.

Rather than official python, Anaconda is installed. 'py.exe' is not available for Anaconda.

**Comment by k-takata** [comment]
Then, I think it's better to use `executable()`.
e.g.
```vim
if executable("py")
  " Use 'py rst2html.py'
elseif executable("python")
  " Use 'python rst2html.py'
else
  " Use 'rst2html.py'
endif

**Comment by mattn** [comment]
Why blocked_ local_ are required?

**Comment by mattn** [comment]
Currently, I'm trying to implement load extra libraries.  #170
Can this PR be applied that way if https://github.com/previm/previm/pull/170 will be merged?

**Comment by mattn** [comment]
#170 is merged. Now you can add extra library by you self.

**Comment by mattn** [comment]
Example to add polyjs

https://github.com/previm/previm/pull/162#issuecomment-1154972196
